hw/ip/otbn/pre_sca/alma/README.md

```diff
@@ -0,0 +1,138 @@
+   `coco-otbn` branch of the tool
+   ```sh
+   git clone git@github.com:abdullahvarici/coco-alma.git -b coco-otbn
+   ```
+   Enter the directory using
+   ```sh
+   cd coco-alma
+   ```
+   Set up a new virtual Python environment
+   ```sh
+   python3 -m venv dev
+   source dev/bin/activate
+   ```
+   And install the Python requirements
+   ```sh
+   pip3 install -r requirements.txt
+   ```
+   Update `examples/otbn/config.json` to point correct locations for `asm`,
+   `objdump` and `rv_objdump`.
+
+1. Generate a Verilog netlist
+
+   A netlist of the DUT can be generated using the Yosys synthesis flow from
+   the OpenTitan repository. From the OpenTitan top level, run
+   ```sh
+   cd hw/ip/otbn/pre_syn
+   ```
+   Set up the synthesis flow as described in the corresponding README. Then run
+   the synthesis
+   ```sh
+   ./syn_yosys.sh
+   ```
+
+## Formally verifying the masking of the OTBN core
+
+After downloading the Alma tool, installing dependencies and synthesizing OTBN,
+the masking can finally be formally verified.
+
+1. Enter the directory where you have downloaded Alma and load the virtual
+   Python environment
+   ```sh
+   source dev/bin/activate
+   ```
+
+1. Make sure to source the `build_consts.sh` script from the OpenTitan
+   repository in order to set up some shell variables.
+   ```sh
+   source ../opentitan/util/build_consts.sh
+   ```
+
+1. Launch the Alma tool to parse, assemble, trace (simulate) and formally verify
+   the netlist. For simplicity, a single script is provided to launch all the
+   required steps with a single command. Simply run
+   ```sh
+   ${REPO_TOP}/hw/ip/otbn/pre_sca/alma/verify_otbn.sh
+   ```
+   This should produce output similar to the one below:
+   ```sh
+   TODO
+   ```
+
+## Individual steps in detail
+
+Below we outline the individual steps performed by the `verify_otbn.sh` script.
+This is useful if you, e.g., want to verify the masking of your own module.
+
+For more details, please refer to the [Alma tutorial](https://github.com/IAIK/coco-alma/tree/hw-verif#usage).
+
+1. Make sure to source the `build_consts.sh` script from the OpenTitan
+   repository in order to set up some shell variables.
+   ```sh
+   source ../opentitan/util/build_consts.sh
+   ```
+
+1. The first step involves the parsing the synthesized netlist.
+   ```sh
+   python3 parse.py --keep --top-module otbn_top_coco \
+      --source ${REPO_TOP}/hw/ip/otbn/pre_sca/alma/rtl/ram_1p.v \
+      ${REPO_TOP}/hw/ip/otbn/pre_sca/alma/rtl/otbn_top_coco.v \
+      ${REPO_TOP}/hw/ip/otbn/pre_syn/syn_out/latest/generated/otbn_core.alma.v
+   ```
+
+1. Next run `assemble.py` script to generate memory initialization file for OTBN.
+   ```sh
+   cd examples/otbn
+   python3 assemble.py --program programs/isw_and.S \
+      --netlist ../../tmp/circuit.v
+   cd ../../
+   ```
+
+1. Then the Verilator testbench can be compiled and run required to identify
+   control signals.
```

Comment:
Then, the Verilator testbench can be compiled and run. This step is required to identify control signals.